### PR TITLE
Missing deleteds attributes names in product stats module 

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Improve `statsproduct::getData`
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/25049
| How to test?      | See https://github.com/PrestaShop/PrestaShop/issues/25049
| Possible impacts? | Product full name is displayed (product + attributes) instead only attributes

### CODE REVIEW

File: statsproduct.php [hunk 1]
```diff
@@ -443,29 +443,10 @@ protected function getData($layers)
...
-                foreach ($product_attribute as $attribute) {
-                    $list .= trim($attribute['group']).' - '.trim($attribute['attr']).', ';
-                }
-                $list = rtrim($list, ', ');
-                $assoc_names[$id_product_attribute] = $list;
-            }
-
-            $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($this->query);
+	    $combinationName = str_replace($product->name, '', $row['product_name']);
             foreach ($result as $row) {
```
Comment (atomiix) [inline statsproduct.php:447]: I think something's wrong here 🤔
You're using `$row` before the loop and use `$result` for the loop but you deleted it just before 😅

**Review by atomiix — CHANGES_REQUESTED** [review]

File: statsproduct.php (hunk 1 above)
Comment (PululuK) [reply statsproduct.php]: 😱

File: statsproduct.php [hunk 2]
```diff
@@ -443,29 +443,11 @@ protected function getData($layers)
...
-                foreach ($product_attribute as $attribute) {
-                    $list .= trim($attribute['group']).' - '.trim($attribute['attr']).', ';
-                }
-                $list = rtrim($list, ', ');
-                $assoc_names[$id_product_attribute] = $list;
-            }
-
-            $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($this->query);
+            $combinationProductNamePrefix = $product->name. '-';
             foreach ($result as $row) {
```
Comment (atomiix) [inline statsproduct.php:447]: `$result` is still not defined at this point 😅

Comment (PululuK) [reply statsproduct.php:448]: 😱

File: statsproduct.php [hunk 3]
```diff
@@ -444,32 +444,33 @@ protected function getData($layers)
...
         }
     }
 
+    /**
+     * @param string $productName
+     * @param string $combinationName
+     *
+     * @return string
+     */
+    protected function removeProductNameInCombinationName(string $productName, string $combinationName): string{
```
Comment (PierreRambaud) [inline statsproduct.php:461]:
```suggestion
    protected function removeProductNameInCombinationName(string $productName, string $combinationName): string
    {
```

File: statsproduct.php [hunk 4]
```diff
@@ -444,32 +444,34 @@ protected function getData($layers)
...
+     *
+     * @return string
+     */
+    protected function removeProductNameInCombinationName(string $productName, string $combinationName): string
+    {
+        $combinationFullNameSepatators = ['-', ':'];
+        $combinationName = str_replace($productName, '', $combinationName);
+        $combinationName = trim($combinationName);
+
+        if(in_array($combinationName[0], $combinationFullNameSepatators)){
```
Comment (PierreRambaud) [inline statsproduct.php:467]: So, if it starts with `-:` the ltrim is only did once, it's the behavior?
Could you please improve the description to better understand your need?

Comment (PululuK) [reply statsproduct.php]: Can you please check https://github.com/PrestaShop/PrestaShop/issues/25049

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @PululuK,

It is ok :heavy_check_mark: 
I checked with 177x branch, 178x branch, and develop branch.
**Without PR**
image
**With PR**
image

Thank you!

**Comment by PierreRambaud** [comment]
Thank you @PululuK

**Pull request merged**